src/routes/transactions/mappers/common/transaction.mapper.ts

```diff
@@ -0,0 +1,142 @@
+        ).submissionDate?.getTime();
+      if (timestamp !== undefined) {
+        prev_page_timestamp = this.getDayInMillis(timestamp, timezoneOffset);
+      } else {
+        throw Error('ExecutionDate cannot be null');
+      }
+      transactionsDomain = transactionsDomain.slice(1);
+    }
+    const transactionsDomainGroups = this.groupByDay(
+      transactionsDomain,
+      timezoneOffset,
+    );
+    transactionsDomainGroups.forEach((transactionGroup) => {
+      if (transactionGroup.timestamp != prev_page_timestamp) {
+        transactionList.push(new DateLabel(transactionGroup.timestamp));
+      }
+      transactionList.push(
+        Promise.all(
+          transactionGroup.transactions.map(async (transaction) => {
+            if (isMultisigTransaction(transaction)) {
+              return new TransactionItem(
+                await this.multisigTransactionMapper.mapTransaction(
+                  chainId,
+                  transaction as MultisigTransaction,
+                  safe,
+                ),
+              );
+            } else if (isModuleTransaction(transaction)) {
+              return new TransactionItem(
+                await this.moduleTransactionMapper.mapTransaction(
+                  chainId,
+                  transaction as ModuleTransaction,
+                  safe,
+                ),
+              );
+            } else if (isEthereumTransaction(transaction)) {
+              const transfers = (transaction as EthereumTransaction).transfers;
+              return transfers
+                ? await Promise.all(
+                    transfers.map(
+                      async (transfer) =>
+                        new TransactionItem(
+                          await this.incomingTransferMapper.mapTransfer(
+                            chainId,
+                            transfer,
+                            safe,
+                          ),
+                        ),
+                    ),
+                  )
+                : null;
+            } else {
+              // This should never happen as AJV would not allow an unknown transaction to get to this stage
+              throw Error('Unrecognized transaction type');
+            }
+          }),
+        ),
+      );
+    });
+
+    return <[TransactionItem | DateLabel]>(
+      (await Promise.all(transactionList)).flat(2)
+    );
+  }
+
+  private groupByDay(
+    transactions: TransactionDomain[],
+    timezoneOffset?: string,
+  ): TransactionDomainGroup[] {
+    return Object.entries(
+      groupBy(transactions, (transaction) => {
+        let transaction_timestamp;
+        if (isMultisigTransaction(transaction)) {
+          transaction_timestamp =
+            transaction.executionDate ?? transaction.submissionDate;
+        } else {
+          transaction_timestamp = transaction.executionDate;
+        }
+        return this.getDayInMillis(
+          transaction_timestamp?.getTime(),
```

Comment:
We are converting here to a UNIX timestamp only to create a date again inside `getDayInMillis`.

Reply:
Done 986baae4b045d33048d9d5c01375019470534632